lib/boolean/boolean.rb

```diff
@@ -1,3 +1,15 @@
 module Boolean; end
-class TrueClass; include Boolean; end
-class FalseClass; include Boolean; end
+class TrueClass
+  include Boolean
+
+  def map(true_value, false_value)
```

Comment:
This idea is interesting! However, there might be a few things I would change. **Please do not take this as discouragement**! I value everyone's contributions and simply want the code to be the best it can possibly be.

1. The name of the method.
  Ruby already has a `map` method which is designed to iterate over an array, returning a new array with the result of the block:

  ```ruby
  arr = [1,2,3,4].map do |i|
    i *2
  end

  puts arr 
  #=> [2, 4, 6, 8]
  ```

  Having a new method named `map` that does not abide by the same contract would be confusing for developers and break some assumptions. I'd recommend changing the name to something not taken.

2. While very OO, these methods seem a tad overkill. I think the same result would be acheived with a conditional or ternary operation, right?

  i.e.
  ```ruby
  some_boolean = true
  (some_boolean) ? 'yes' : 'no'
  #=> 'yes'
  ```

  vs your implementation: 

  ```ruby
  some_boolean = true
  some_boolean.map('yes', 'no')
  #=> 'yes'
  ```

  While your solution is better from an object oriented point of view, I'd say that it makes the developer question the code. They will need to know what the `map` method is and how it works. All of this will be buried in our gem here and may seem like magic to them. On the other hand, a conditional expression is core to the language and understood by all.